### PR TITLE
#5 - Blog app - controllers specs

### PR DESCRIPTION
**Pull request by Amtresu**
- [x] Add RSpec as a gem to your project.
- [x] Create a Request spec file for all your controllers.
- [x] Make sure that for each action you check:
- [x] If response status was correct.
- [x] If a correct template was rendered.
- [x] If the response body includes correct placeholder text.

### CODE REVIEW

**Review by Reem-lab — COMMENTED** [review]
# Hi @Amtresu, 👋🏻 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, I think that you submit the wrong link to the wrong activity so kindly make sure to submit the right link since this is link for the views milestobe but you are making the controllers milestone 💯🚀✨

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/controllers/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      expect(response).to have_http_status :ok
+    end
+
+    it 'Should show the index template' do
+      expect(response).to render_template :show
+    end
+
+    it 'Should incldue the correct HTML' do
+      expect(response.body).to include '<div class="post_detail">'
+    end
```
Comment (codepantha) [inline spec/controllers/posts_spec.rb:31]: - [ ] The highlighted tests are failing. Kindly see the image attached below and try to make the failing tests green ✔️ 
image

File: spec/controllers/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      expect(response).to have_http_status 200
+    end
+
+    it 'Should show the index template' do
+      expect(response).to render_template :show
+    end
+
+    it 'Should incldue the correct HTML' do
+      expect(response.body).to include '<div class="user_detail_div">'
+    end
```
Comment (codepantha) [inline spec/controllers/users_spec.rb:31]: - [ ] The following tests for the `show` action of the User are not passing. Please see the image attached below and try to make all the tests green ✔️ 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Amtresu 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow
✔️ Controller specs
✔️ Professional README
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Amtresu** [comment]
Hi @codepantha These tests are passing on my end 
image
So I will need more information about why they are failing for you to try and correct these.

**Review by addod19 — APPROVED** [review]
### Hi @Amtresu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [x] Once all the test passed on your end and you have shown that with a screenshot :+1: we are good to go :rocket: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**